frontend_lib/src/component/ShareDownload/NewShareDownload.jsx

```diff
@@ -0,0 +1,153 @@
+        </div>
+
+        <div className='shareDownload__email'>
+          <textarea
+            type='text'
+            className='shareDownload__email__input form-control'
+            placeholder={props.t('Enter the email address of the recipient(s)')}
+            rows='10'
+            value={props.shareEmails}
+            onChange={props.onChangeEmails}
+            onKeyDown={props.onKeyDownEnter}
+          />
+          <button
+            type='button'
+            className='shareDownload__email__icon'
+            id='popoverMultipleEmails'
+            key='share_emails'
+            style={{':hover': {color: props.hexcolor}}}
+          >
+            <i className='fa fa-fw fa-question-circle' />
+          </button>
+          <Popover placement='bottom' isOpen={state.popoverOpen} target='popoverMultipleEmails' toggle={this.popoverToggle}>
+            <PopoverBody>{props.t('To add multiple recipients, separate the email addresses with a comma or space.')}</PopoverBody>
+          </Popover>
+        </div>
+
+        {state.passwordActive
+        ? <div className='shareDownload__password'>
+          <div className='shareDownload__password__wrapper'>
+            <i className='fa fa-fw fa-lock' />
+            <input
+              type={state.hidePassword ? 'password' : 'text'}
+              className='shareDownload__password__input form-control'
+              placeholder={props.t('Password to share link (optional)')}
+              value={props.sharePassword}
+              onChange={props.onChangePassword}
+              onFocus={props.onKeyDownEnter}
+            />
+            <button
+              type='button'
+              className='shareDownload__password__icon'
+              key='seeSharePassword'
+              title={props.t('Show password')}
+              style={{':hover': {color: props.hexcolor}}}
+              data-cy='seePassword'
+              onClick={this.handleSeePassword}
+            >
+              <i className={state.hidePassword ? 'fa fa-fw fa-eye' : 'fa fa-fw fa-eye-slash'} />
+            </button>
+          </div>
+          <button
+            type='button'
+            className='shareDownload__password__icon'
+            key='randomSharePassword'
+            title={props.t('Generate random password')}
+            style={{':hover': {color: props.hexcolor}}}
+            onClick={this.handleRandomPassword}
+          >
+            <i className='fa fa-fw fa-repeat' />
+          </button>
+        </div>
+        : <div className='shareDownload__password'>
+          <span className='shareDownload__password__link' onClick={this.handleSeePasswordInput}>
+            {props.t('Protect by password')}
+          </span>
+        </div>
+        }
+        <div className='d-flex mt-3'>
+          <button
+            className='shareDownload__cancel btn outlineTextBtn'
+            key='cancelNewShare'
+            onClick={props.onClickReturnToManagement}
```

Comment:
the handle should be called handleClickCancelButton or handleClickCancelNewShare